cli/module_reload.go

```diff
@@ -17,14 +23,72 @@ import (
 // Using maps directly also saves a lot of high-maintenance ser/des work.
 type ModuleMap map[string]any
 
-// configureModule is the configuration step of module reloading. Returns (needsRestart, error).
+// ServiceMap is the same kind of thing as ModuleMap (see above), a map representing a single service.
+type ServiceMap map[string]any
+
+// addShellService adds a shell service to the services slice if missing. Mutates part.RobotConfig.
+func addShellService(c *cli.Context, vc *viamClient, part *apppb.RobotPart, wait bool) error {
+	partMap := part.RobotConfig.AsMap()
+	if _, ok := partMap["services"]; !ok {
+		partMap["services"] = make([]any, 0, 1)
+	}
+	services, _ := rutils.MapOver(partMap["services"].([]any), //nolint:errcheck
+		func(raw any) (ServiceMap, error) { return ServiceMap(raw.(map[string]any)), nil },
+	)
+	if slices.ContainsFunc(services, func(service ServiceMap) bool { return service["type"] == "shell" }) {
+		debugf(c.App.Writer, c.Bool(debugFlag), "shell service found on target machine, not installing")
+		return nil
+	}
+	services = append(services, ServiceMap{"name": "shell", "type": "shell"})
+	asAny, _ := rutils.MapOver(services, func(service ServiceMap) (any, error) { //nolint:errcheck
+		return map[string]any(service), nil
+	})
+	partMap["services"] = asAny
+	if err := writeBackConfig(part, partMap); err != nil {
+		return err
+	}
+	infof(c.App.Writer, "installing shell service on target machine for file transfer")
+	if err := vc.updateRobotPart(part, partMap); err != nil {
+		return err
+	}
+	if !wait {
+		return nil
+	}
+	// note: we wait up to 7 seconds; that's 5 seconds for the config watcher and 2 for luck.
+	// If we don't do this, reloading fails on first run.
```

Comment:
Ah this is interesting + clever. But, @cheukt is there any better way we can ensure that the shell service has been added and started (apart from sleeping)? There's none that I can think of.

Reply:
semi related, would be nice to have either an RPC that tells RDK to fetch configs, or use a stream to notify about new configs

Reply:
I don't think there's a better way currently. Config watcher is 10s tho I think

Reply:
oops -- updated loop to wait 11s